meta-client/src/main/java/org/apache/helix/metaclient/recipes/leaderelection/LeaderElectionListenerInterface.java

```diff
@@ -0,0 +1,32 @@
+package org.apache.helix.metaclient.recipes.leaderelection;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * It provides APIs for listener listening on events like a new leader is created or current
+ * leader node is deleted.
+ */
+public interface LeaderElectionListenerInterface {
+  // When new leader is elected:
+  //                             noLeader (null)                  ->    has leader (new leader name)
+  // When existing leader not leader anymore:
```

Comment:
Question: does this mean the leader history will be persisted? Because if the leader is gone, usually we dont know which was prev leader.

Reply:
If we register data listener on a particular entry, we will get an event for 'node is deleted' event when leader (node) is gone. We will know prev leader -> no leader change.

Reply:
Yes. We will present leader info in memory.